pkg/comp-functions/functions/common/pdb.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"context"
+	"fmt"
+
+	pdbv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	fnproto "github.com/crossplane/function-sdk-go/proto/v1beta1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+)
+
+func AddPDBSettings(comp Composite) func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
+	return func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
+
+		log := svc.Log
+
+		err := svc.GetObservedComposite(comp)
+		if err != nil {
+			return runtime.NewFatalResult(fmt.Errorf("can't get composite: %w", err))
+		}
+
+		log.Info("Checking if PDB is needed", "service", comp.GetName(), "instances", comp.GetInstances())
+
+		if comp.GetInstances() < 2 {
+			return runtime.NewNormalResult("Not HA, no pdb needed")
+		}
+		log.Info("HA detected, adding pdb", "service", comp.GetName())
+
+		min := intstr.IntOrString{IntVal: int32(comp.GetInstances()) / 2}
```

Comment:
If the string doesn't work, can we at least make sure that we always round up here? Then we'll have the same functionality as using `50%`

Reply:
ok, I found the issue with `50%`, Type: intval is default one, when setting a strVal: "50%", You must add explicitly Type: strVal